### PR TITLE
Sets up interface and incorporates option models for BEHAVIOR!

### PR DESCRIPTION
**Pull request by NishanthJKumar**
Sets up interface (i.e, `_BehaviorOptionModel` in `option_models.py`) as well as option models (included in `behavior_options.py`) for BEHAVIOR!

Planning time is significantly faster (though still cannot solve test task due to sampler/other issues that will be addressed in a separate PR). Bottleneck is no longer `env.step()` calls, but rather time between option model executions (roughly 1 min between each action).

### CODE REVIEW

File: src/envs/behavior.py
Comment (ronuchit) [inline src/envs/behavior.py:457]: you'll need to add memory_controller here too with your bit being True

Comment (NishanthJKumar) [reply src/envs/behavior.py:457]: right, but that's not implemented yet, so i wasn't sure whether to include this in this PR

Comment (NishanthJKumar) [reply src/envs/behavior.py:457]: should i add the option model code to this PR as well? Or should i open a separate PR that includes that stuff in `behavior_options.py` first and merge that first?

File: src/option_model.py
Comment (ronuchit) [inline src/option_model.py:60]: change to `option.memory.get["model_controller"]` so that if the key doesn't exist, we'll hit this assert

File: src/option_model.py
Comment (ronuchit) [inline src/option_model.py:61]: `final_state` -> `next_state`

File: src/envs/behavior.py
Comment (ronuchit) [inline src/envs/behavior.py:439]: this assert can be written as `assert memory.get("policy_controller") is not None`

File: src/envs/behavior.py
Comment (ronuchit) [reply src/envs/behavior.py:457]: i'd say add it to this PR. but you can do the changing of the argument type from iGibsonEnv to BehaviorEnv in a separate PR

File: src/envs/behavior.py
Comment (ronuchit) [inline src/envs/behavior.py:448]: formatting is a bit weird, thanks yapf -_- can you move this comment to its own line directly preceding the line of code, so that  `assert memory.get("policy_controller") is not None` fits in one line?

File: src/envs/behavior.py
Comment (ronuchit) [inline src/envs/behavior.py:475]: great to see that this worked!

File: src/envs/behavior_options.py
Comment (ronuchit) [inline src/envs/behavior_options.py:1044]: can you go through and make sure all the docstrings are correct now? for instance, the one for this method says "If the placement is infeasible, returns an indication to this effect (None). Otherwise, returns a function that can be stepped like an option to output actions at each timestep." but this is out-of-date, because now we have the possibility that it can return a model as well. can you describe in the comment exactly what's going on with this crazy-looking return type? (for all the controller fns, not just this one)

**Review by ronuchit — APPROVED** [review]
glad this ended up being quite a simple change at the end of the day! my only major thing is to go through the comments and (a) make sure they're correct, (b) describe exactly what the crazy-looking return types are for the controller_fns! let @tomsilver review before merging.

File: src/envs/behavior.py
Comment (tomsilver) [inline src/envs/behavior.py:470]: if I understand correctly, this is going to run planning twice -- once in creating the policy controller, and once in creating the model controller. is that right? can we avoid that if so?

this is making me want more the refactor where we avoid the controller functions altogether, and just create options directly in `behavior_options.py`... but if there's an easy way to address this concern in the current paradigm that's fine too, and maybe I or someone else can try the refactor in a later PR.

File: src/envs/behavior_options.py
Comment (tomsilver) [reply src/envs/behavior_options.py:1044]: +1

**Review by tomsilver — COMMENTED** [review]
looking good thanks! one other small thing is I don't love the name "model_controller", but I can't think of a better alternative

File: src/envs/behavior.py
Comment (NishanthJKumar) [reply src/envs/behavior.py:470]: Yes, that is exactly right (and something I discovered only after running it lol).

I don't really know an easy way to address this in the current paradigm unfortunately... Would love to refactor to create options directly, though not sure if we should merge this and do that in a separate PR?

Comment (tomsilver) [reply src/envs/behavior.py:470]: I'll leave it up to you (or @ronuchit if he feels strongly). If there's any chance that this refactor could also help clear up the bugs you're seeing with the current code, that would be reason to prioritize the refactor. But it sounds like you've localized the issue to the grasp sampler, so it might not.

Comment (ronuchit) [reply src/envs/behavior.py:470]: can we just make each controller fn cache the results of the planning/RRT part? that way it can be re-used?

**Pull request merged**
